torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp

```diff
@@ -1007,6 +1007,49 @@ RegisterOperators reg2({
           push(stack, std::move(t));
         },
         aliasAnalysisFromSchema()),
+    Operator(
+        "aten::sum.int(int[] self) -> int",
+        [](Stack* stack) {
+          c10::List<int64_t> l = pop(stack).toIntList();
+          auto sum = 0;
+          for (const auto& elem : l) {
+            sum += elem;
+          }
+          push(stack, sum);
+        },
+        aliasAnalysisFromSchema()),
+    Operator(
+        "aten::sum.float(float[] self) -> float",
+        [](Stack* stack) {
+          c10::List<double> l = pop(stack).toDoubleList();
+          auto sum = 0.0;
+          for (const auto& elem : l) {
+            sum += elem;
+          }
+
+          // To handle cases where the List only contains int,
```

Comment:
OOC, why is this special case needed?

Reply:
This is to handle cases where the the List has been annotated as `float` but it only contains `integer` elements. Then the sum of all the elements of the list will also be `int`. So this case handles that. Here is the example. 

```
def (x : List[float]):
       return sum(x)
x = [1, 2, 3]
```

Reply:
if we have a int list inside a float[] operator i think that's a bug that should be fixed elsewhere.

Reply:
@eellison : So ideally, we should make the float to a int list then?

Reply:
Ok, confirmed with @eellison offline and we should not support int list inside float[]. I have removed this part of the code snippet in the new version. Please review.